src/components/AlbumForm.vue

```diff
@@ -434,19 +426,17 @@ export default {
         }
       });
 
-      Promise.all(promises).then(() => {
-        let promise = null;
-        if (this.album) {
-          promise = this.update({ id: this.album.id, newAlbum: transformed });
-        } else {
-          promise = this.create(transformed);
-        }
-        promise.then((succeeded) => {
-          if (succeeded) {
-            this.$router.push(this.$route.query.redirect || { name: "albums" });
-          }
-        });
-      });
+      await Promise.all([...artist_map, ...label_map]);
+      let promise = null;
```

Comment:
`promise` feels like a bad name here. The whole point is that we're moving away from promises (should also be fixed in a lot of places).

Reply:
IMHO promise is still the correct term for a async function that is not yet resolved. `async/await` is another way of resolving promises.
But if you have another suggestion, I'll gladly change it

Reply:
`pendingResult`?